From c5121f35f8551254dd428aaccaab76fc399ad66b Mon Sep 17 00:00:00 2001
From: Tong Wu <tong1.wu@intel.com>
Date: Tue, 12 Oct 2021 18:18:37 +0800
Subject: [PATCH 2/2] qsv: enable d3d11va support for YUV422 and YUV444

Make 8-bit, 10-bit, 12-bit YUV 4:2:2 video sources as well as YUV 4:4:4
video sources supported by d3d11va.

Signed-off-by: Tong Wu <tong1.wu@intel.com>
---
 libavcodec/qsv.c          | 4 ++--
 libavutil/hwcontext_qsv.c | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/libavcodec/qsv.c b/libavcodec/qsv.c
index 9fda80b2dc..96ebbb0283 100644
--- a/libavcodec/qsv.c
+++ b/libavcodec/qsv.c
@@ -207,7 +207,7 @@ enum AVPixelFormat ff_qsv_map_fourcc(uint32_t fourcc)
     case MFX_FOURCC_NV12: return AV_PIX_FMT_NV12;
     case MFX_FOURCC_P010: return AV_PIX_FMT_P010;
     case MFX_FOURCC_P8:   return AV_PIX_FMT_PAL8;
-#if CONFIG_VAAPI
+#if CONFIG_VAAPI || CONFIG_D3D11VA
     case MFX_FOURCC_YUY2: return AV_PIX_FMT_YUYV422;
 #if QSV_VERSION_ATLEAST(1, 17)
     case MFX_FOURCC_AYUV: return AV_PIX_FMT_0YUV;
@@ -238,7 +238,7 @@ int ff_qsv_map_pixfmt(enum AVPixelFormat format, uint32_t *fourcc)
     case AV_PIX_FMT_P010:
         *fourcc = MFX_FOURCC_P010;
         return AV_PIX_FMT_P010;
-#if CONFIG_VAAPI
+#if CONFIG_VAAPI || CONFIG_D3D11VA
 #if QSV_VERSION_ATLEAST(1, 9)
     case AV_PIX_FMT_BGRA:
         *fourcc = MFX_FOURCC_RGB4;
diff --git a/libavutil/hwcontext_qsv.c b/libavutil/hwcontext_qsv.c
index b19559b757..fe6b426466 100644
--- a/libavutil/hwcontext_qsv.c
+++ b/libavutil/hwcontext_qsv.c
@@ -108,7 +108,7 @@ static const struct {
     { AV_PIX_FMT_BGRA, MFX_FOURCC_RGB4 },
     { AV_PIX_FMT_P010, MFX_FOURCC_P010 },
     { AV_PIX_FMT_PAL8, MFX_FOURCC_P8   },
-#if CONFIG_VAAPI
+#if CONFIG_VAAPI || CONFIG_D3D11VA
     { AV_PIX_FMT_YUYV422,
                        MFX_FOURCC_YUY2 },
 #if QSV_VERSION_ATLEAST(1, 17)
@@ -1185,7 +1185,7 @@ static int map_frame_to_surface(const AVFrame *frame, mfxFrameSurface1 *surface)
         surface->Data.R = frame->data[0] + 2;
         surface->Data.A = frame->data[0] + 3;
         break;
-#if CONFIG_VAAPI
+#if CONFIG_VAAPI || CONFIG_D3D11VA
     case AV_PIX_FMT_YUYV422:
         surface->Data.Y = frame->data[0];
         surface->Data.U = frame->data[0] + 1;
-- 
2.16.1.windows.4

